### PR TITLE
Macro to simplify Decimal creation

### PR DESCRIPTION
**Pull request by r001**
Created the dec! macro to simplify Decimal creation.
```rust
dec!(5).to_string() = "5"
dec!(1.23).to_string() = won't compile
dec!("1.23").to_string() = "1.23"
dec!(123, -2).to_string() = "1.23"
dec!(123, 2).to_string() = "12300"
dec!(170_141_183_460_469_231_732, 0) = won't compile
dec!(1, 4_294_967_296) = panic!("Overflow of arg2.")
```

### CODE REVIEW

File: scrypto/src/types/decimal.rs [hunk 1]
```diff
@@ -99,6 +99,46 @@ from_int!(i64);
...
+       Decimal::from($x) 
+    };
+    
+    ($int:literal, $exponent:literal) => {
+        if ($exponent) < 0 {
+            Decimal(($int) * PRECISION)
+                .div(10i128.pow((-1i128 * ($exponent)) as u32))
+        } else {
+            Decimal(($int) * PRECISION)
+                .mul(10i128.pow((1i128 * ($exponent)) as u32))
```
Comment (iamyulong) [inline scrypto/src/types/decimal.rs:137]: `as u32` will silence underflow/overflow instead of panic. I'd suggest we force a check before doing the calculation and add a test case to cover this edge case.

**Review by iamyulong — COMMENTED** [review]
Good work! Left one comment below. Please also update the PR description :)

File: scrypto/src/types/decimal.rs (hunk 1 above)
Comment (r001) [reply scrypto/src/types/decimal.rs]: @iamyulong I have updated the code according to your remarks. Please review again.

**Review by iamyulong — APPROVED** [review]
